docs/protocol.md

```diff
@@ -186,12 +187,125 @@ and its arguments (`params`).
 
 Here are the different methods that can be called:
 
+* Greeting
 * Subscribe
 * Unsubscribe
 * Catchup
 * Broadcast
 * Publish
 
+### Greeting new clients
+🧭 **RPC Message** > **Query** > **Greeting**
+
+This is a message that is sent to clients after they have connected to a server.
```

Comment:
This is incoherent with the JSON-Schema specification.
When I connect to a server, I don't know (yet) which lao_id I'm connecting to.
Yet the `params` object contains a `channel`. 
We can either send this message only after the client `subscribe`s to a valid LAO, _or_ exclude the `channel` from the `params`.

Reply:
Conversely, if it is indeed related to a `channel`, why put it at the JSON-RPC level and not encapsulated in a usual `Message` object ? (I'm not sure if it'd be a good idea, just asking a possibly-dumb question)

Reply:
I guess it also depends on whether we want a greeting message per LAO or per organizer server. Do we want to support the case where a given server hosts multiple LAOs and the set of peers/witnesses is different for each lao?
I would argue yes, and thus I think it actually might make more sense to have this message being sent on the lao channel?

Regarding why it is  at the JSON-RPC level, I do not know haha. I just assumed that there was a good reason for doing so when PR #846 was created 😂

Reply:
@Ajkunas @romain46 @ahominal @Robingoumaz
What do you think? Per LAO (including the `channel` parameter to prevent replays, at least I think that this is its purpose)? And should we pack it into a message instead of using a different `method`?

Reply:
Doesn't the QR code that the organizer supplies contain the IP address of the server + LAO ID? If so, would make sense to send LAO ID along as a type of "Server Name Indication" like in SSL.

Sounds to me that maybe it should be encapsulated in the message under the `subscribe`  method _only_ when subscribing to the overall LAO ID.

Reply:
Yes I think that makes sense! So we just send the greeting on the lao channel after subscribing and replace the channel field with LAO ID?

Reply:
What I am not sure about is the subscribe method since this message originates from the server 🤔 A `broadcast` of a `publish` might be more fitting?

Reply:
Something like this?

https://github.com/dedis/popstellar/blob/efef6116a71dab7854b76305003bdd5603230d4f/protocol/query/method/message/data/dataGreetLao.json#L1-L49

Reply:
Looks great!

Reply:
This is from a discussion during the frontend meeting in the week before the easter break:

For now there is no way we can determine what backend is the organizer's since we could also connect through witnesses servers and the greeting message does not necessarily have to come from the organizer. Determining who the organizer is, is important for the secret ballot since (for now) only the organizer should be able to send election keys and the frontends need to verify the signature and check whether it was indeed the organizer who signed the message.

To do this we can bind the organizer's backend public key to the organizer's frontend public key (which is part of the lao#create message). Since the same might be required for the witnesses in the future, just treating the lao#create message as a one-sided claim and then requiring a witness signature might not be ideal.

We could for example add another field to the greeting message called `frontend` where the backend claims that it is the backend of some frontend public key. The corresponding frontend could then either just sign this message using the witnessing feature or send a separate message to everyone in order to finish the bidirectional claim. When going with the witnessing approach, a given lao#greet message would only be valid with the signature of the frontend. (And since the corresponding frontend is usually the first to connect to its backend, this signature will usually already be there)
With this approach, backends can be bound to the corresponding frontends and it is possible to determine the organizers backend. What are your thoughts on this?

Reply:
Also I just realized that we do not need the `sender` field, this value can be retrieved from the layer above :)